harness/determined/pytorch/dsat/_dsat_search_method.py

```diff
@@ -320,28 +330,32 @@ def gpu_mem(self) -> int:
         assert (
             self.model_profile_info_trial is not None
         ), "The model profile info Trial must be run before calling this method."
-        return self.model_profile_info_trial.metric["gpu_mem"]
```

Comment:
`self.model_profile_info_trial.metric` will always be a dict and if `gpu_mem` is a missing key, that should just be a `KeyError` since something has gone very wrong.

Reply:
Agreed